packages/m2m-gateway/test/api/consumerDelegations/createConsumerDelegation.test.ts

```diff
@@ -59,39 +61,50 @@ describe("POST /consumerDelegations router test", () => {
     expect(res.status).toBe(403);
   });
 
-  it("Should return 400 if passed an invalid delegation seed", async () => {
-    const token = generateToken(authRole.M2M_ADMIN_ROLE);
-    const res = await makeRequest(token, {
-      invalidParam: "invalidValue",
-    });
-
-    expect(res.status).toBe(400);
-  });
-
-  it("Should return 500 in case of missingMetadata error", async () => {
-    mockDelegationService.createConsumerDelegation = vi
-      .fn()
-      .mockRejectedValue(missingMetadata());
-    const token = generateToken(authRole.M2M_ADMIN_ROLE);
-    const res = await makeRequest(token, mockDelegationSeed);
+  it.each([
+    { invalidParam: "invalidValue" },
+    { eserviceId: mockDelegationSeed.eserviceId },
+    { delegateId: mockDelegationSeed.delegateId },
+    { eserviceId: "invalidId" },
+    { delegateId: "invalidId" },
```

Comment:
```suggestion
    { ...mockDelegationSeed, invalidParam: "invalidValue" },
    { eserviceId: mockDelegationSeed.eserviceId },
    { delegateId: mockDelegationSeed.delegateId },
    { ...mockDelegationSeed, eserviceId: "invalidId" },
    { ...mockDelegationSeed, delegateId: "invalidId" },
```
I suggest to pass the mockSeed in order to let it fail precisely for these cases, otherwise it also fails because it doesn't have all fields